smtp.go

```diff
@@ -167,12 +170,67 @@ func (c *smtpSender) Send(from string, to []string, msg io.WriterTo) error {
 		return err
 	}
 
+	r := &smtpRecorder{&bytes.Buffer{}}
+	defer r.Record(c)()
```

Comment:
Had to read this twice to get that it is not the actual recording that is defered. Maybe this is worth splitting up?

```
done := r.Record(c)
defer done()
```